contracts/ERC721TokenRegistry.sol

```diff
@@ -0,0 +1,193 @@
+pragma solidity 0.4.18;
+
+import "zeppelin-solidity/contracts/ownership/Ownable.sol";
+
+
+/**
+ * The ERC721TokenRegistry is a basic registry mapping token symbols
+ * to their known, deployed addresses on the current blockchain.
+ *
+ * Note that the TokenRegistry does *not* mediate any of the
+ * core protocol's business logic, but, rather, is a helpful
+ * utility for Terms Contracts to use in encoding, decoding, and
+ * resolving the addresses of currently deployed tokens.
+ *
+ * At this point in time, administration of the Token Registry is
+ * under Dharma Labs' control.  With more sophisticated decentralized
+ * governance mechanisms, we intend to shift ownership of this utility
+ * contract to the Dharma community.
+ */
+contract ERC721TokenRegistry is Ownable {
+    mapping (bytes32 => TokenAttributes) public symbolHashToTokenAttributes;
+    string[4294967296] public tokenSymbolList;
```

Comment:
worth documenting that you can't have dynamically sized arrays of strings in solidity? thus the crazy allocation of memory...